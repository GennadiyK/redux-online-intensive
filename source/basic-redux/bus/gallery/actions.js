import { SHOW_NEXT_PHOTO, SHOW_SELECTED_PHOTO } from './types'
export const showNextPhoto = {
  type: SHOW_NEXT_PHOTO,
};

export const showSelectedPhoto = (index) => {
  return {
    type: SHOW_SELECTED_PHOTO,
    payload: index,
  }
}