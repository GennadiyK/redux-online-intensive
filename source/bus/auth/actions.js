import { types } from './types';

export const authAction = {
    authenticate: () => {
        return {
            type: types.AUTHENTICATE,
        }
    },
    signupAsync: ( userData )=> {
        return {
            type: types.SIGNUP_ASYNC,
            payload: userData,
        };
    },
};