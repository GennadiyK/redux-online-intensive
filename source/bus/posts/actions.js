import { types } from './types';
import { api } from '../../REST/index';

export const postsActions = {
    fillPosts: (posts) => {
        return {
            type:    types.FILL_POSTS,
            payload: posts,
        };
    },
    createPost: (post) => {
        return {
            type:   types.CREATE_POST,
            payload: post,
        };
    },
    fetchPosts: () => async (dispatch) => {
        dispatch({
            type: types.FETCH_POSTS,
        });
    },
    clearPosts: () => {
        return {
            type: types.CLEAR_POSTS,
        }
    },
    removePost: (id) => {
        return {
            type: types.REMOVE_POST,
            payload: id
        }
    },


    createPostAsync: (comment) => {
        return {
            type: types.CREATE_POST_ASYNC,
            payload: comment,
        }
    },
    removePostAsync: (id) => {
        return {
            type: types.REMOVE_POST_ASYNC,
            payload: id,
        }
    },
}
