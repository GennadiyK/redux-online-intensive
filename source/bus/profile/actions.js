import { types } from './types';

export const profileActions = {
    fillProfile: (profile)=> {
        return {
            type: types.FILL_PROFILE,
            payload: profile,
        };
    },
    cleaProfile: ()=> {
        return {
            type: types.CLEAR_PROFILE,
        };
    },
    updateAvatar: (newAvatarUrl)=> {
        return {
            type: types.UPDATE_AVATAR,
            payload: newAvatarUrl,
        };
    },
    updateNameAsync: (newName)=> {
        return {
            type: types.UPDATE_NAME_ASYNC,
            payload: newName,
        };
    },
    updateAvatarAsync: (newAvatarUrl)=> {
        return {
            type: types.UPDATE_AVATAR_ASYNC,
            payload: newAvatarUrl,
        };
    },
    updatePasswordAsync: (passwordData)=> {
        return {
            type: types.UPDATE_PASSWORD_ASYNC,
            payload: passwordData,
        };
    },
};