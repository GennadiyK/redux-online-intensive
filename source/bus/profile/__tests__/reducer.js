import { fromJS, List } from "immutable";
import { profileActions } from "../actions";
import { profileRducer } from "../reducer";

const initialState = List();

describe("profile reducer", () => {
  test("show initial state by default", () => {
    expect(profileRducer(void 0, {})).toMatchInlineSnapshot(`
Immutable.Map {
  "id": "",
  "firstName": "",
  "lastName": "",
  "avatar": "",
  "token": "",
}
`);
  });

  test("show handle FILL_PROFILE", () => {
    expect(profileRducer(void 0, profileActions.fillProfile(__.userProfile)))
      .toMatchInlineSnapshot(`
Immutable.Map {
  "id": "TEST_ID",
  "firstName": "Walter",
  "lastName": "White",
  "avatar": "TEST_AVATAR",
  "token": "TEST_TOKEN",
}
`);
  });
  test("show handle UPDATE_AVATAR", () => {
    expect(profileRducer(void 0, profileActions.updateAvatar(__.url)))
      .toMatchInlineSnapshot(`
Immutable.Map {
  "id": "",
  "firstName": "",
  "lastName": "",
  "avatar": "https://www.url.com",
  "token": "",
}
`);
  });

  test("show handle CLEAR_PROFILE", () => {
    expect(
      profileRducer(void 0, profileActions.cleaProfile())
    ).toMatchInlineSnapshot(`Immutable.Map {}`);
  });
});
