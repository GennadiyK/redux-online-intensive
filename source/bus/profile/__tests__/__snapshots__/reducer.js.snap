// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile reducer show initial state by default 1`] = `
Immutable.Map {
  "id": "",
  "firstName": "",
  "lastName": "",
  "avatar": "",
  "token": "",
}
`;
